### PR TITLE
python3Packages.packaging: 20.9 -> 21.3

### PR DESCRIPTION
**Pull request by jyooru**
###### Motivation for this change

#144678

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/packaging/default.nix [hunk 1]
```diff
@@ -10,12 +10,12 @@
 
 buildPythonPackage rec {
   pname = "packaging";
-  version = "20.9";
+  version = "21.2";
   format = "pyproject";
```
Comment (fabaff) [inline pkgs/development/python-modules/packaging/default.nix:14]:
```suggestion
  version = "21.2";
  format = "pyproject";
  
  disabled = pythonOlder "3.6";
```

see https://github.com/pypa/packaging/blob/main/setup.py#L50

**Review by fabaff — COMMENTED** [review]
This needs to go to `staging` instead of `master`.

- `six` seems to be not longer a [requirement](https://github.com/pypa/packaging/blob/main/setup.py#L51).
- Please add `pythonImportsCheck`
- `checkInputs` could be removed as the tests are not executed

**Comment by jonringer** [comment]
This is a breaking staging change. We will need to wait til after the next staging-next cycle to merge it:

https://discourse.nixos.org/t/lets-have-a-great-21-11-release-cycle/15262/3?

**Comment by jyooru** [comment]
#152550

**Pull request closed without merging**